pkg/genevalogging/genevalogging.go

```diff
@@ -0,0 +1,547 @@
+package genevalogging
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
+	securityclient "github.com/openshift/client-go/security/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/tls"
+)
+
+const (
+	kubeNamespace      = "openshift-azure-logging"
+	kubeServiceAccount = "system:serviceaccount:" + kubeNamespace + ":geneva"
+
+	fluentbitImage = "arosvc.azurecr.io/fluentbit:1.3.9-1" //"docker.io/fluent/fluent-bit:0.12.19"
+	mdsdImage      = "arosvc.azurecr.io/genevamdsd:master_249"
+
+	parsersConf = `
+[PARSER]
+	Name containerpath
+	Format regex
+	Regex ^/var/log/containers/(?<POD>[^_]+)_(?<NAMESPACE>[^_]+)_(?<CONTAINER>.+)-(?<CONTAINER_ID>[0-9a-f]{64})\.log$
+
+[PARSER]
+	Name crio
+	Format regex
+	Regex ^(?<TIMESTAMP>[^ ]+) [^ ]+ [^ ]+ (?<MESSAGE>.*)$
+	Time_Key TIMESTAMP
+	Time_Format %Y-%m-%dT%H:%M:%S.%L
+`
+
+	journalConf = `
+[INPUT]
+	Name systemd
+	Tag journald
+	DB /var/lib/fluent/journald
+
+[OUTPUT]
+	Name forward
+`
+
+	containersConf = `
+[SERVICE]
+	Parsers_File /etc/td-agent-bit/parsers.conf
+
+[INPUT]
+	Name tail
+	Path /var/log/containers/*
+	Path_Key path
+	Tag journald
+	DB /var/lib/fluent/containers
+	Parser crio
+
+[FILTER]
+	Name parser
+	Match journald
+	Key_Name path
+	Parser containerpath
+	Reserve_Data true
+
+[FILTER]
+	Name grep
+	Match journald
+	Regex NAMESPACE ^(?:default|kube-.*|openshift|openshift-.*)$
+
+[OUTPUT]
+	Name forward
```

Comment:
[nit] I'd prefer setting the forward port explicitly either in cmd line -o parameter or in the configuration here. It would make it easier to understand how mdsd and fluent bit are connecting to each other.